typescript/demo/function-call-stream.ts

```diff
@@ -217,18 +209,20 @@ async function functionCallingWithAIConfig() {
       return;
     }
 
-    const message = output.data as ChatCompletionMessageParam;
+    const rawResponse = output.metadata?.rawResponse;
+    const function_call = rawResponse?.function_call;
```

Comment:
I don't think this is a good model. If the only way for someone to support function calling is by delving into the raw response, then we've missed a critical scenario.

Reply:
Ok after rebasing onto https://github.com/lastmile-ai/aiconfig/pull/628, I will update but do this at the end of the diff stack. I just want to keep this for string-only to make it easier to review, then the rest of diffs will be easier to followup